# Phylogenetic Biology - Final Project

## Guidelines - you can delete this section before submission

This repository is a stub for your final project. Fork it, develop your project, and submit it as a pull request. Edit/ delete the text in this readme as needed.

Some guidelines and tips:

- Use the stubs below to write up your final project. Alternatively, if you would like the writeup to be an executable document (with [knitr](http://yihui.name/knitr/), [jupytr](http://jupyter.org/), or other tools), you can create it as a separate file and put a link to it here in the readme.

- For information on formatting text files with markdown, see https://guides.github.com/features/mastering-markdown/ . You can use markdown to include images in this document by linking to files in the repository, eg `![GitHub Logo](/images/logo.png)`.

- The project must be entirely reproducible. In addition to the results, the repository must include all the data (or links to data) and code needed to reproduce the results.

- If you are working with unpublished data that you would prefer not to publicly share at this time, please contact me to discuss options. In most cases, the data can be anonymized in a way that putting them in a public repo does not compromise your other goals.

- Paste references (including urls) into the reference section, and cite them with the general format (Smith at al. 2003).

- Commit and push often as you work.

OK, here we go.

# Title of my project

Extinction and the loss of phylogenetic diversity

## Introduction and Goals

It seems inevitable that climate change, along with other anthropogenic factors, will continue to drive the endangerment and extinction of taxa for the immediate future. Each species is by definition a unique parcel of genetic information, which means that an extinction is the loss of a unique genome. With the number of endgangered taxa mounting, it is important to understand the ways in which species will go extinct, and what that means for the loss of phylogenetic diversity. 

__**The methods I will use to do this are...**__

My goal is to construct the phylogeny for amphibians and map the degree to which each tip node is imperiled onto the phylogeny. With this information, along with a relatedness metric such as  distance between tip node sequence, I hope to observe the pattern of endangerment within the phylogenetic tree (whether it is clustered or uniform), and possibly even which types of genetic information (genes) are most frequently lost. 



__**The data I will use are (my own data/ data publicly available at YYY/ simulations)**__

I expect I will draw on data from an existing published phylogeny or collection of published data. I will score degree of endangerment by classification on IUCN red list, though I will look into a continuous variable that describes endangerment. 

___________

Davies, T. Jonathan, Gideon F. Smith, Dirk U. Bellstedt, James S. Boatwright, Benny Bytebier, Richard M. Cowling, Félix Forest, Luke J. Harmon, A. Muthama Muasya, Brian D. Schrire, Yolande Steenkamp, Michelle Van Der Bank, and Vincent Savolainen. "Extinction Risk and Diversification Are Linked in a Plant Biodiversity Hotspot." PLoS Biology PLoS Biol 9.5 (2011): n. pag. Web. <http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1000620>.

Purvis, A. "Nonrandom Extinction and the Loss of Evolutionary History." Science 288.5464 (2000): 328-30. Web. <http://science.sciencemag.org/content/288/5464/328>.

Rolland, J., M. W. Cadotte, J. Davies, V. Devictor, S. Lavergne, N. Mouquet, S. Pavoine, A. Rodrigues, W. Thuiller, L. Turcati, M. Winter, L. Zupan, F. Jabot, and H. Morlon. "Using Phylogenies in Conservation: New Perspectives." Biology Letters 8.5 (2011): 692-94. Web. <http://rsbl.royalsocietypublishing.org/content/early/2011/11/21/rsbl.2011.1024>.

Thuiller, Wilfried, Sébastien Lavergne, Cristina Roquet, Isabelle Boulangeat, Bruno Lafourcade, and Miguel. B. Araujo. "Consequences of Climate Change on the Tree of Life in Europe." Nature 470.7335 (2011): 531-34. Web. <http://www.nature.com/nature/journal/v470/n7335/full/nature09705.html>.

____________

## Methods

I started by downloading the accession numbers for 12 genes published Pyron 2011, used in the construction of a comprehensive Amphibia phylogeny, storing them in a data frame called accessions.  

```{r}
setwd("~/Dropbox/Brown University/BIOL1425/phylobio_final_project/")
accessions=read.csv("Pyrondata.csv",header = TRUE)
# data is from Pyron 2011
row.names(accessions)=accessions[,1]
specieslist=accessions[,1]
accessions=accessions[-2873,]
accessions=accessions[,-1]
dim(accessions)
accessions[1,]

```
I then downloaded a list of all the IUCN designations from the IUCN Red List database (http://www.iucnredlist.org/), storing it locally as a .csv and within the R environment as IUCN_amphibia.
```{r}
IUCN_amphibia=read.csv("IUCN_amphibia.csv", header = TRUE)
IUCN_amphibia
dim(IUCN_amphibia)
names(IUCN_amphibia)
row.names(IUCN_amphibia)=IUCN_amphibia[,6]
IUCN_amphibia=IUCN_amphibia[,-6]
```
By searching the row names of each dataframe, I pared down the rows in accessions so that I only look at species with designations in the IUCN database. This removed about 500 rows from the original Pyron dataset.
```{r}
accessions=accessions[rownames(accessions)%in%rownames(IUCN_amphibia),]
dim(accessions)
accessions[1,]
#filtered accessions so that only contains rows for species with values listed in IUCN_amphibia
```
I then created an IUCN_amphibia_matched dataframe, which listed only those rows which are in accessions, effectively creating two dataframes with identical rows.
```{r}
IUCN_amphibia_matched=IUCN_amphibia[rownames(IUCN_amphibia)%in%rownames(accessions),]
IUCN_amphibia_matched
dim(IUCN_amphibia_matched)
names(IUCN_amphibia_matched)
```
I then went into the matched IUCN dataframe and extracted the Red_list_status and Red_list_criteria columns and appended them to my accessions dataframe. 
```{r}
accessions[,13]=IUCN_amphibia_matched[,9]
accessions[,14]=IUCN_amphibia_matched[,10]
names(accessions)[13]="Red_list_status"
names(accessions)[14]="Red_list_criteria"
names(accessions)


#x=row.names(accessions[which(!is.na(accessions[,1])==TRUE),])
#the species for which there are values!=NA in column 1
#y=accessions[which(!is.na(accessions[,1])==TRUE),1]
#the values for the species that have values in column 1


#y=accessions[which(!is.na(accessions[,1])==TRUE),1]

#which(!is.na(accessions[,1])==TRUE)
#row.names(accessions[which(!is.na(accessions[,1])==TRUE),])
#accessions[which(!is.na(accessions[,1])==TRUE),]

```
I then exported my accessions dataframe to a .csv file
```{r}
write.csv(accessions,file = "accessions2.csv")
```

The tools I used were... See analysis files at (links to analysis files).



## Results

The tree in Figure 1...

## Discussion

These results indicate...

The biggest difficulty in implementing these analyses was...

If I did these analyses again, I would...

## References


