# Phylogenetic Biology - Final Project

## Guidelines - you can delete this section before submission

This repository is a stub for your final project. Fork it, develop your project, and submit it as a pull request. Edit/ delete the text in this readme as needed.

Some guidelines and tips:

- For information on formatting text files with markdown, see https://guides.github.com/features/mastering-markdown/ . You can use markdown to include images in this document by linking to files in the repository, eg `![GitHub Logo](/images/logo.png)`.

- Paste references (including urls) into the reference section, and cite them with the general format (Smith at al. 2003).

OK, here we go.
```{r}
setwd("~/Dropbox/Brown University/BIOL1425/phylobio_final_project/")
```

# Title of my project

Sensitivity Analysis of input GE on EDGE Values

## Introduction and Goals

It seems inevitable that climate change, along with other anthropogenic factors, will continue to drive the endangerment and extinction of taxa for the immediate future. Each species is by definition a unique parcel of genetic information, which means that an extinction is the loss of a unique genome. With the number of endgangered taxa mounting, it is important to understand the ways in which species will go extinct, and what that means for the loss of phylogenetic diversity. EDGE values (Isaac et al 2007) are a proposed method of combining a species' probability of extinction (GE, global endangerment) with a measure of its unique evolutionary history (ED, evolutionary distinctiveness). Calculation of the evolutionary distinctiveness (ED) value is based on tree topology, where a species' score is  the total branch length from the root node to its tip, where internal branches' weight is scaled by the number of tip taxa that share that branch.

![Isaac2007_fig1](Isaac_fig1.png)


__**The methods I will use to do this are...**__

My goal is to construct the phylogeny for Salamandridae and map EDGE values (evolutionarily distinct, globally endangered) onto the tips.  


__**The data I will use are (my own data/ data publicly available at YYY/ simulations)**__

I am drawing my data entirely from the supplementary table of accession numbers provided in Pyron 2011.

____________

## Methods

I started by downloading the accession numbers for 12 genes published Pyron 2011, used in the construction of a comprehensive Amphibia phylogeny, storing them in a data frame called accessions.  

```{r}
accessions=read.csv("Pyrondata.csv",header = TRUE)
# data is from Pyron 2011
row.names(accessions)=accessions[,1]
specieslist=accessions[,1]
accessions=accessions[-2873,]
accessions=accessions[,-1]
dim(accessions)
accessions[1,]

```
I then downloaded a list of all the IUCN designations for species within Salamandridae from the IUCN Red List database (http://www.iucnredlist.org/), storing it locally as a .csv and within the R environment as IUCN_salamandridae.
```{r}
IUCN_salamandridae=read.csv("IUCN_salamandridae.csv", header = TRUE)
dim(IUCN_salamandridae)
#76 species of salamanders in IUCN database
names(IUCN_salamandridae)
row.names(IUCN_salamandridae)=IUCN_salamandridae[,4]
IUCN_salamandridae=IUCN_salamandridae[,-4]
```
By searching the row names of each dataframe, I pared down the rows in accessions so that I only look at species with designations in the IUCN database. This left me with 65 salamandridae species for phylogenetic analysis.
```{r}
accessions=accessions[rownames(accessions)%in%rownames(IUCN_salamandridae),]
dim(accessions)
accessions[1,]
```
After looking at the accessions for each listed species, it became clear that the salamanders are data deficient for all genes except x12s, x16s, and cytb. For that reason, I dropped the other 9 genes from my analysis.
```{r}
accessions=accessions[,-c(4:12)]
```
I then created an IUCN_salamandridae_matched dataframe, which listed only those rows which are in accessions, effectively creating two dataframes with identical rows.
```{r}
IUCN_salamandridae_matched=IUCN_salamandridae[rownames(IUCN_salamandridae)%in%rownames(accessions),]
dim(IUCN_salamandridae_matched)
names(IUCN_salamandridae_matched)
```
I then went into the matched IUCN dataframe and extracted the Red_list_status and Red_list_criteria columns and appended them to my accessions dataframe. 
```{r}
accessions[,4]=IUCN_salamandridae_matched[,7]
accessions[,5]=IUCN_salamandridae_matched[,8]
accessions[,6]=IUCN_salamandridae_matched[,11]
names(accessions)[4]="Red_list_status"
names(accessions)[5]="Red_list_criteria"
names(accessions)[6]="population_trend"
names(accessions)
dim(accessions)

```
I then exported my accessions dataframe to a .csv file
```{r}
write.csv(accessions,file = "accessions_salamandridae.csv")
```
I removed T.vittatus sequence from x12s.fasta because it would not align (tried inputting the reverse complement without success).
```{r}
sala.concat.tree=read.tree("sala_data/concat.reduced_tree/RAxML_bestTree.concat.reduced_boot100")
plot(sala.concat.tree,cex = 0.5)
sala.concat.tree=clade.matrix(sala.concat.tree)
ED_sala.concat=ed.calc(sala.concat.tree)
ED_sala.concat=data.frame(ED_sala.concat$spp)
ED_sala.concat=ED_sala.concat[order(ED_sala.concat[,1]),]
IUCN_salamandridae_matched=IUCN_salamandridae_matched[rownames(IUCN_salamandridae_matched)%in%ED_sala.concat[,1],]

ED_sala.concat=ED_sala.concat[ED_sala.concat[,1]%in%rownames(IUCN_salamandridae_matched),]

ED_sala.concat[,3:4]=IUCN_salamandridae_matched[,c(7,11)]

get.EDGE=function(matrix,model){
  matrix[,5]=NA
  if (model=="Isaac"){
    matrix[which(matrix[,3]=="CR"),5]=log(matrix[which(matrix[,3]=="CR"),2]*.4)
    matrix[which(matrix[,3]=="EN"),5]=log(matrix[which(matrix[,3]=="EN"),2]*.2)
    matrix[which(matrix[,3]=="VU"),5]=log(matrix[which(matrix[,3]=="VU"),2]*.1)
    matrix[which(matrix[,3]=="NT"),5]=log(matrix[which(matrix[,3]=="NT"),2]*.05)
    matrix[which(matrix[,3]=="LC"),5]=log(matrix[which(matrix[,3]=="LC"),2]*.025)
  }
  if (model=="IUCN50"){
    matrix[which(matrix[,3]=="CR"),5]=log(matrix[which(matrix[,3]=="CR"),2]*.97)
    matrix[which(matrix[,3]=="EN"),5]=log(matrix[which(matrix[,3]=="EN"),2]*.42)
    matrix[which(matrix[,3]=="VU"),5]=log(matrix[which(matrix[,3]=="VU"),2]*.05)
    matrix[which(matrix[,3]=="NT"),5]=log(matrix[which(matrix[,3]=="NT"),2]*.004)
    matrix[which(matrix[,3]=="LC"),5]=log(matrix[which(matrix[,3]=="LC"),2]*.00005)
  }
  if (model=="IUCN100"){
    matrix[which(matrix[,3]=="CR"),5]=log(matrix[which(matrix[,3]=="CR"),2]*.999)
    matrix[which(matrix[,3]=="EN"),5]=log(matrix[which(matrix[,3]=="EN"),2]*.667)
    matrix[which(matrix[,3]=="VU"),5]=log(matrix[which(matrix[,3]=="VU"),2]*.1)
    matrix[which(matrix[,3]=="NT"),5]=log(matrix[which(matrix[,3]=="NT"),2]*.01)
    matrix[which(matrix[,3]=="LC"),5]=log(matrix[which(matrix[,3]=="LC"),2]*.0001)
  }
  if (model=="IUCN500"){
    matrix[which(matrix[,3]=="CR"),5]=log(matrix[which(matrix[,3]=="CR"),2]*1)
    matrix[which(matrix[,3]=="EN"),5]=log(matrix[which(matrix[,3]=="EN"),2]*.996)
    matrix[which(matrix[,3]=="VU"),5]=log(matrix[which(matrix[,3]=="VU"),2]*.39)
    matrix[which(matrix[,3]=="NT"),5]=log(matrix[which(matrix[,3]=="NT"),2]*.02)
    matrix[which(matrix[,3]=="LC"),5]=log(matrix[which(matrix[,3]=="LC"),2]*.0005)
  }
  matrix[,5]=abs(matrix[,5])
  colnames(matrix)[5]="EDGE"
  matrix=matrix[order(matrix[,5],decreasing = TRUE),]
  matrix[6]=1:dim(matrix)[1]
  colnames(matrix)[6]="rank by edge"
  return(matrix[,c(1,5,6)])
}
sala.concat.Isaac=get.EDGE(matrix = ED_sala.concat,"Isaac")
sala.concat.50=get.EDGE(matrix = ED_sala.concat,model = "IUCN50")
sala.concat.100=get.EDGE(matrix = ED_sala.concat,"IUCN100")
sala.concat.500=get.EDGE(matrix = ED_sala.concat,model = "IUCN500")
#get edges

#top 5 from each transformation by EDGE value
sala.concat.Isaac[1:5,]
sala.concat.50[1:5,]
sala.concat.100[1:5,]
sala.concat.500[1:5,]


rank.distance=function(matrix1,matrix2){
  counter<-0
  for (i in 1:dim(matrix1)[1]){
    pairdiff<-abs(matrix1[i,3]-matrix2[which(matrix2[,1]==matrix1[i,1]),3])
    counter<-counter+pairdiff
  }
  print(counter)
}
rank.distance(sala.concat.Isaac,sala.concat.Isaac)
rank.distance(sala.concat.Isaac,sala.concat.50)
rank.distance(sala.concat.Isaac,sala.concat.100)
rank.distance(sala.concat.Isaac,sala.concat.500)

rank.distance(sala.concat.50,sala.concat.Isaac)
rank.distance(sala.concat.50,sala.concat.50)
rank.distance(sala.concat.50,sala.concat.100)
rank.distance(sala.concat.50,sala.concat.500)

rank.distance(sala.concat.100,sala.concat.Isaac)
rank.distance(sala.concat.100,sala.concat.50)
rank.distance(sala.concat.100,sala.concat.100)
rank.distance(sala.concat.100,sala.concat.500)

rank.distance(sala.concat.500,sala.concat.Isaac)
rank.distance(sala.concat.500,sala.concat.50)
rank.distance(sala.concat.500,sala.concat.100)
rank.distance(sala.concat.500,sala.concat.500)

```
The tools I used were... 

See analysis files at (links to analysis files).

Accession numbers for sequences were gathered from Pyron 2011
mafft alignment
raxml (starting from hydroid)
  justification in Pyron for ml method
  
## Results

The tree in Figure 1...

## Discussion

These results indicate...

The biggest difficulty in implementing these analyses was...

If I did these analyses again, I would...

## References

Davies, T. Jonathan, Gideon F. Smith, Dirk U. Bellstedt, James S. Boatwright, Benny Bytebier, Richard M. Cowling, Félix Forest, Luke J. Harmon, A. Muthama Muasya, Brian D. Schrire, Yolande Steenkamp, Michelle Van Der Bank, and Vincent Savolainen. "Extinction Risk and Diversification Are Linked in a Plant Biodiversity Hotspot." PLoS Biology PLoS Biol 9.5 (2011): n. pag. Web. <http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1000620>.

Isaac, Nick J.b., Samuel T. Turvey, Ben Collen, Carly Waterman, and Jonathan E.m. Baillie. "Mammals on the EDGE: Conservation Priorities Based on Threat and Phylogeny." PLoS ONE 2.3 (2007): n. pag. Web. 28 Apr. 2016. <http://journals.plos.org/plosone/article/asset?id=10.1371/journal.pone.0000296.PDF>.

Purvis, A. "Nonrandom Extinction and the Loss of Evolutionary History." Science 288.5464 (2000): 328-30. Web. <http://science.sciencemag.org/content/288/5464/328>.

Pyron, R. Alexander, and John J. Wiens. "A Large-scale Phylogeny of Amphibia including over 2800 Species, and a Revised Classification of Extant Frogs, Salamanders, and Caecilians." Molecular Phylogenetics and Evolution 61.2 (2011): 543-83. Web. 28 Apr. 2016. <http://www.sciencedirect.com/science/article/pii/S105579031100279X>.

Rolland, J., M. W. Cadotte, J. Davies, V. Devictor, S. Lavergne, N. Mouquet, S. Pavoine, A. Rodrigues, W. Thuiller, L. Turcati, M. Winter, L. Zupan, F. Jabot, and H. Morlon. "Using Phylogenies in Conservation: New Perspectives." Biology Letters 8.5 (2011): 692-94. Web. <http://rsbl.royalsocietypublishing.org/content/early/2011/11/21/rsbl.2011.1024>.

Thuiller, Wilfried, Sébastien Lavergne, Cristina Roquet, Isabelle Boulangeat, Bruno Lafourcade, and Miguel. B. Araujo. "Consequences of Climate Change on the Tree of Life in Europe." Nature 470.7335 (2011): 531-34. Web. <http://www.nature.com/nature/journal/v470/n7335/full/nature09705.html>.


